### PR TITLE
Fixed some small bugs and added input validation

### PR DESCRIPTION
**Pull request by TimMcGilly**
Closes https://github.com/RadiationChallengeProjectCAST/RadDataStorage/issues/15

Add a lot better errors messages

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -53,121 +59,142 @@ app.post('/api/upload_data', async (req, res) => {
...
+                code: 102,
+                message: 'Invalid token.',
+            }],
+        });
         return;
     }
-
     const teamID = result.rows[0].teamid;
 
     const insertQuery = 'INSERT INTO reading (teamid, posfloor, posx, posy, cpm) VALUES ($1, $2, $3, $4, $5);';
```
Comment (tomkettle1) [inline index.js:84]: Is this the query that was playing up @TimMcGilly? If so, what did you say the error was?

Comment (TimMcGilly) [reply index.js:84]: @tomkettle1 Maybe, there were issues with this in the past due to wrong order of COMMIT and ROLLBACK which was fixed in previous  commit. Also changing row count here fixes a different bug. Think it works now.

File: Dockerfile [hunk 2]
```diff
@@ -2,7 +2,7 @@ FROM node:12.12.0-alpine
 
 COPY . .
 RUN apk --no-cache add curl
-RUN chmod +x wait-for-db.sh
+RUN sed -i 's/\r$//' wait-for-db.sh  && chmod +x wait-for-db.sh
```
Comment (dislocated-su) [inline Dockerfile:5]: What exactly is going on here?

**Review by tomkettle1 — APPROVED** [review]
Can't see any issues. I like the move away from the json structure for holding location. All the SQL stuff looks fine now. The only thing I'd say is for logging errors to the console, do we get a timestamp? It might be useful to also give the user of the API the timestamp so if they report the issue we can better find their error

File: Dockerfile (hunk 2 above)
Comment (TimMcGilly) [reply Dockerfile:5]: Fixes line endings on file

**Review by dislocated-su — APPROVED** [review]

**Pull request merged**